pkg/ssh/keymanager.go

```diff
@@ -23,26 +22,35 @@ import (
 
 const (
 	// SSHKeySize is the size of the SSH key.
-	SSHKeySize = 4096
+	SSHKeySize     = 4096
+	KnownHostsFile = "pdc_known_hosts"
 )
 
 // KeyManager manages SSH keys and certificates. It ensures that the SSH keys,
 // certificates and known_hosts files exist in their configured locations.
+// EnsureCertExists should only be called once the service has started successfully.
+
+type KeyManager interface {
+	services.Service
+	EnsureCertExists(bool) error
+}
+
+// pdcKeyManager implements KeyManager. If needed, it gets new certificates signed
+// by the PDC API.
+//
 // If the service starts successfully, then the key and cert files will exist.
 // It will attempt to reuse existing keys and certs if they exist.
-type KeyManager struct {
+type pdcKeyManager struct {
 	*services.BasicService
```

Comment:
I am wondering if the KeyManager really needs to be a dskit Service ? There is no background job / looking running process that must be running. I would suggest keeping this a regular struct. WDYT?

Reply:
That would make it simpler. I'll do that.